.github/CHANGELOG.md

```diff
@@ -1,3 +1,15 @@
+## Release 0.4.1 (development release)
+
+### Bug fixes
+
+* Updates required `pydantic` version from 1.8.2 to 2.4, while still continuing to use "v1" features. [#46](https://github.com/XanaduAI/xanadu-cloud-client/pull/46)
```

Comment:
Pedantic, but I think any version of Pydantic v2 should be compatible with these changes.

Also, this comment can mirror the one in the v0.3.1 release notes:
```suggestion
* Set lower bound on compatible `pydantic` versions to v2.0.0. [(#46)](https://github.com/XanaduAI/xanadu-cloud-client/pull/46)
```